mmv1/third_party/terraform/resources/resource_sql_database_instance.go.erb

```diff
@@ -285,28 +278,34 @@ settings.backup_configuration.binary_log_enabled are both set to true.`,
 										Optional:     true,
 										Set:          schema.HashResource(sqlDatabaseAuthorizedNetWorkSchemaElem),
 										Elem:         sqlDatabaseAuthorizedNetWorkSchemaElem,
-										AtLeastOneOf: ipConfigurationKeys,
+										AtLeastOneOf: []string{"settings.0.ip_configuration.0.authorized_networks", "settings.0.ip_configuration.0.ipv4_enabled", "settings.0.ip_configuration.0.require_ssl", "settings.0.ip_configuration.0.private_network"},
 									},
 									"ipv4_enabled": {
 										Type:         schema.TypeBool,
 										Optional:     true,
 										Default:      true,
-										AtLeastOneOf: ipConfigurationKeys,
+										AtLeastOneOf: []string{"settings.0.ip_configuration.0.authorized_networks", "settings.0.ip_configuration.0.ipv4_enabled", "settings.0.ip_configuration.0.require_ssl", "settings.0.ip_configuration.0.private_network"},
 										Description:  `Whether this Cloud SQL instance should be assigned a public IPV4 address. At least ipv4_enabled must be enabled or a private_network must be configured.`,
 									},
 									"require_ssl": {
 										Type:         schema.TypeBool,
 										Optional:     true,
-										AtLeastOneOf: ipConfigurationKeys,
+										AtLeastOneOf: []string{"settings.0.ip_configuration.0.authorized_networks", "settings.0.ip_configuration.0.ipv4_enabled", "settings.0.ip_configuration.0.require_ssl", "settings.0.ip_configuration.0.private_network"},
 									},
 									"private_network": {
 										Type:             schema.TypeString,
 										Optional:         true,
 										ValidateFunc:     orEmpty(validateRegexp(privateNetworkLinkRegex)),
 										DiffSuppressFunc: compareSelfLinkRelativePaths,
-										AtLeastOneOf:     ipConfigurationKeys,
+										AtLeastOneOf:     []string{"settings.0.ip_configuration.0.authorized_networks", "settings.0.ip_configuration.0.ipv4_enabled", "settings.0.ip_configuration.0.require_ssl", "settings.0.ip_configuration.0.private_network"},
 										Description:      `The VPC network from which the Cloud SQL instance is accessible for private IP. For example, projects/myProject/global/networks/default. Specifying a network enables private IP. At least ipv4_enabled must be enabled or a private_network must be configured. This setting can be updated, but it cannot be removed after it is set.`,
 									},
+									"allocated_ip_range": {
```

Comment:
If this isn't specified, is one automatically assigned?

Reply:
If so, the field should be `optional `+ `computed`, right? I currently did not add the `computed` and also not specify a value in the test and it did not throw me any errors, so I assume it means the field is not optional computed?

Reply:
Given this field is tied to using a private network, I think we'd need a test that uses a private network without setting `allocated_ip_range` to verify that. We don't have one right now (note, `TestAccSqlDatabaseInstance_settings_checkServiceNetworking` looks like a case where we do so, but the config is intentionally invalid so it won't successfully apply.)

Reply:
Sure! I will add one test then.